### PR TITLE
CI: Rebase code with main branch when running the action

### PR DESCRIPTION
**Pull request by eloycoto**
On PR rebase the current code with the main branch, if there is any
    conflict will fail, but code will run against the target branch.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -19,6 +19,16 @@ jobs:
...
+      - name: Fetch target branch and rebase it
+        if: github.event_name == 'pull_request_target'
+        run: |
+          set -x
+          git config user.email "CI@action.git"
+          git config user.name "CI"
+          git remote add upstream https://github.com/${{ github.repository }}.git
+          git fetch upstream ${{ github.event.pull_request.base.ref }}
+          git rebase upstream/${{ github.event.pull_request.base.ref }}
+
```
Comment (masayag) [inline .github/workflows/test.yml:31]: I'm surprised there isn't action ready for that. I'd imagine this should be pretty common.

Comment (eloycoto) [reply .github/workflows/test.yml:31]: not at all, there is the pull_request_target event..

**Pull request merged**